src/cli/expand-patterns.js

```diff
@@ -68,9 +68,9 @@ async function* expandPatternsInternal(context) {
 
     const stat = await lstatSafe(absolutePath);
     if (stat) {
-      if (stat.isSymbolicLink()) {
+      if (stat.isSymbolicLink() && context.argv.errorOnUnmatchedPattern !== false) {
         yield {
-          error: `Explicitly specified pattern "${pattern}" is a symbolic link.`,
+          error: `Skipping pattern "${pattern}", as it points to a symbolic link.`,
         };
```

Comment:
The request was to avoid throwing an error if `--no-error-on-unmatched-pattern` is set, not to continue processing the file as if it were regular.